crawl-ref/source/describe.cc

```diff
@@ -2967,7 +2967,9 @@ static string _player_spell_desc(spell_type spell)
 
     if (!you_can_memorise(spell))
     {
-        description += "\nYou cannot memorise this spell because "
+        description += (string) "\nYou cannot "
```

Comment:
I'm confused as to why you need the cast here. What compiler version and on what system?

`desc_cannot_memorise_reason` has type string, so the compiler should be able to figure out that you're building a  `string` to append to description. 

That said, it would probably be better to re-factor this whole method to use an `ostringstream` instead of a bunch of implicit string appends with `+` and `+=`.

Reply:
Before appending things to `description`, the code first concatenates "\nYou cannot" with the result of `you.has_spell(spell) ? "cast" : "memorise", neither of which are of type `string`, and so the `+` operation fails (for me at least). If one of the first two operands is explicitly a `string`, then it is able to use the built-in `+` operator to form a `string`.

(I'm using gcc version 7.3.0)

Reply:
Oh, I see.

ostringstreaem is not `+`ed, it is `<<`ed, and has a method `.str()` to return a string. So you'd do something like

```
ostringstream description;

description << "Spells and stuff mp cost: " << 3;

return description.str()
```

Reply:
Got it, thanks!